examples/vnet/doca_dpdk.c

```diff
@@ -568,39 +572,54 @@ struct endecap_layer doca_endecap_layers[] = {
 	{FILL_GRE_HDR, doca_dpdk_build_gre_header},
 };
 
-static void doca_dpdk_build_endecap_data(uint8_t **header,
+static void doca_dpdk_build_raw_data(uint8_t **header,
 					 struct doca_flow_pipe_cfg *cfg,
-					 uint16_t flags)
+					 uint16_t flags, uint8_t type)
 {
 	uint8_t idx;
 	struct endecap_layer *layer;
 
 	for (idx = 0; idx < RTE_DIM(doca_endecap_layers); idx++) {
 		layer = &doca_endecap_layers[idx];
 		if (flags & layer->layer)
-			layer->fill_data(header, cfg);
+			layer->fill_data(header, cfg, type);
 	}
 }
 
+static void doca_dpdk_build_encap_action(struct doca_dpdk_action_entry *entry,
+										 struct doca_flow_pipe_cfg *cfg, uint8_t layer)
+{
+	uint8_t *header;
+	struct rte_flow_action *action = entry->action;
+	struct doca_dpdk_action_rawencap_data *encap;
+
+	encap = &entry->action_data.rawencap;
+	header = encap->data;
```

Comment:
could not find who inits this data

Reply:
no init, we build it in doca_dpdk_build_raw_data